### ISSUE TITLE
Yandex Browser on Android doesn't support uBO, but it works fine

### ISSUE THREAD

**Issue by hawkeye116477** [yandex/browser-extensions#9]
Welcome! When I want to install uBlock Origin(`https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm?hl=pl` ) from Chrome Web Store on Yandex Stable on Android, then it tells that it doesn't support this extension. However if I install it in developer mode (`https://github.com/yandex/browser-extensions/wiki/Custom-Extension-Install-In-Developer-Mode)`, then it works perfectly, hiding and blocking what I want, better than content blocking api (of course I disabled native blocking when I tested uBO). So why uBO is blocked from installing from Chrome Web Store?

**Comment by yafz** [comment]
The blame is on Yandex from what I've heard. They are explicitly hindering plugin installations.

**Comment by sdoeringNew** [comment]
In the newest versions the workaround with the developer mode is not working anymore.

First install an older version of the "Yandex Browser with Protect", add uBlock Origin as described above, and then you can update Yandex at your Play Store.

I've found an old APK at APKMirror, e.g.: https://www.apkmirror.com/apk/%d1%8f%d0%bd%d0%b4%d0%b5%d0%ba%d1%81/yandex-browser/yandex-browser-19-6-3-318-release/#downloads

**Comment by lethinhrider** [comment]
> Welcome! When I want to install uBlock Origin(`https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm?hl=pl` ) from Chrome Web Store on Yandex Stable on Android, then it tells that it doesn't support this extension. However if I install it in developer mode (`https://github.com/yandex/browser-extensions/wiki/Custom-Extension-Install-In-Developer-Mode)`, then it works perfectly, hiding and blocking what I want, better than content blocking api (of course I disabled native blocking when I tested uBO). So why uBO is blocked from installing from Chrome Web Store?


The solution is to use the official dev build of uBlock Origin: https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii
I have successfully installed it on Yandex android browser.

**Comment by LifeIsStrange** [comment]
Thanks, by the way do you know a way of knowing which chromium version is yandex alpha based on?
(yandex alpha should have better extension support, it has a dedicated menu, etc)
@lethinhrider

**Comment by pavor84** [comment]
> Thanks, by the way do you know a way of knowing which chromium version is yandex alpha based on?
> (yandex alpha should have better extension support, it has a dedicated menu, etc)
> @lethinhrider

YandexBrowser passes chromium version in its user agent, like "Chrome/84.0.4147.20"

**Comment by LifeIsStrange** [comment]
@pavor84 you're right. Html5test.com failed to parse it but it's definitely in the user agent

**Comment by inspire22** [comment]
It's showing "an error has occurred" when installing the main ublock, and the dev build link above shows "not compatible" instead of "install" - any tips for 2021 please?  The whole reason I was trying yandex was to get ublock working :)

**Comment by Kippykip** [comment]
> > Welcome! When I want to install uBlock Origin(`https://chrome.google.com/webstore/detail/ublock-origin/cjpalhdlnbpafiamejdnhcphjbkeiagm?hl=pl` ) from Chrome Web Store on Yandex Stable on Android, then it tells that it doesn't support this extension. However if I install it in developer mode (`https://github.com/yandex/browser-extensions/wiki/Custom-Extension-Install-In-Developer-Mode)`, then it works perfectly, hiding and blocking what I want, better than content blocking api (of course I disabled native blocking when I tested uBO). So why uBO is blocked from installing from Chrome Web Store?
> 
> The solution is to use the official dev build of uBlock Origin: https://chrome.google.com/webstore/detail/ublock-origin-development/cgbcahbpdhpcegmbfconppldiemgcoii
> I have successfully installed it on Yandex android browser.

Looks like this workaround doesn't work as of recently. Even when I fake the user agent string with a newer version of chrome it still says unsupported. Welp i guess ill try the apk downgrade method

**Comment by jaytaylor** [comment]
The developer ublock origin build is working for Android Yandex as of late February, 2022.

**Comment by lethinhrider** [comment]
I installed Yandex browser on my android smartphone from 2 years ago.  Then installed the dev version of uBlock Origin and it's been working fine so far.  But I still hope Yandex browser can install the stable version of uBlock Origin since it's easy to find in Chrome Store.
 Does anyone know why Yandex browser can't install uBlock Origin stable version since 2 years ago?

**Comment by madranet** [comment]
I've not been able to install any unpacked extensions in developer mode lately. I get that  infuriating 'manifest file is missing... ' error every time. I'll have to try the dev build of uBlock and see if I have any luck with that. I can't stand using the internet without uBlock.

I really don't know why Yandex have decided to break one of the two standout features [the other being text-reflow] that used to make this the only browser on Android worth using. Once I've enabled developer mode just let me load whatever extensions I want, FFS! If they don't function properly, or they break something, then it's my problem for trying something that didn't work. Why have a developer mode in the first place, if you're still going to cripple it?
